pkg/cue/model/value/value.go

```diff
@@ -82,6 +83,30 @@ func (val *Value) UnmarshalTo(x interface{}) error {
 	return json.Unmarshal(data, x)
 }
 
+// SubstituteInStruct substitute expr in struct lit value
+// nolint:staticcheck
+func (val *Value) SubstituteInStruct(new ast.Expr, key string) error {
+	node := val.CueValue().Syntax(cue.ResolveReferences(true))
+	x, ok := node.(*ast.StructLit)
+	if !ok {
+		return errors.New("value is not a struct lit")
+	}
+	for i := range x.Elts {
```

Comment:
Does this mean that only the outermost values can be modified?

Reply:
Yes I think we can optimize that later since the usage is very limited currently.